sql/sql_acl.cc

```diff
@@ -14312,6 +14313,22 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else if (redirect_enabled)
+  {
+    char msg[MAX_REDIRECTION_LEN];
+    size_t len = 0;
+
+    msg[0] = '\0';
+    len = snprintf(msg + 1, MAX_REDIRECTION_LEN - 1, "Location: mysql://[%s]:%s/user=%s&ttl=%s\n", 
+      redirect_server_host, redirect_server_port, sctx->user, redirect_server_ttl);
+
+    if (len >= MAX_REDIRECTION_LEN - 1){
+      sql_print_error("redirection info is too large to return to client (len= %u)",len);
+      my_ok(thd);
+    }
+    else
+      my_ok(thd, 0, 0, msg);
+  }
   else
     my_ok(thd);
 
```

Comment:
you have my_ok as last statement  3  times , in all branches. One is enough, I think . Simplify it, please.

Reply:
Removed redundant my_ok().